crates/types/src/net/cluster_state.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2023 - 2025 Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use super::{define_rpc, TargetName};
+use serde::{Deserialize, Serialize};
+
+define_rpc! {
+    @request= NodePing,
+    @response= NodePong,
+    @request_target=TargetName::NodePing,
+    @response_target=TargetName::NodePong,
+}
+
+/// Gossip Push message. Is pushed from each node to every other known node
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct NodePing {
+    //todo(azmy): share latest metadata versions?
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct NodePong {}
```

Comment:
What do you imagine will be in response and do we actually need a response?

Reply:
I see your point :) `Pongs` are definitely not needed as their own message type and can be dropped. The failure detector still treat it as a ping anyway.